Sprint-Mission1/js/login.js

```diff
@@ -7,90 +7,88 @@ document.addEventListener("DOMContentLoaded", function () {
   const closeEyesbtn = document.querySelector(".close-eyes");
   const openEyesbtn = document.querySelector(".open-eyes");
 
+  let unEmailValid = false;
+  let unPasswordValid = false;
 
+  // 에러 메시지 출력 함수
+  function showError(inputElement, errorElement, errorMessage) {
+    errorElement.textContent = errorMessage;
+    inputElement.classList.add("error");
+  }
+
+  // 에러 메시지 지우기 함수
+  function clearError(inputElement, errorElement) {
+    errorElement.textContent = "";
+    inputElement.classList.remove("error");
+  }
 
   // 이메일 유효성 검사
   function validateEmail() {
     const emailValue = emailInput.value.trim();
-    emailError.textContent = "";
-    emailInput.classList.remove("error");
+    clearError(emailInput, emailError);
 
     if (!emailValue) {
-      emailError.textContent = "이메일을 입력해주세요.";
-      emailInput.classList.add("error");
-      return false;
+      showError(emailInput, emailError, "이메일을 입력해주세요.");
+      unEmailValid = false;
     } else if (!/^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(emailValue)) {
-      emailError.textContent = "잘못된 이메일 형식입니다.";
-      emailInput.classList.add("error");
-      return false;
+      showError(emailInput, emailError, "잘못된 이메일 형식입니다.");
+      unEmailValid = false;
+    } else {
+      unEmailValid = true;
     }
-    return true;
+    toggleLoginButton();
   }
 
   // 비밀번호 유효성 검사
   function validatePassword() {
     const passwordValue = passwordInput.value.trim();
-    passwordError.textContent = "";
-    passwordInput.classList.remove("error");
+    clearError(passwordInput, passwordError);
 
     if (!passwordValue) {
-      passwordError.textContent = "비밀번호를 입력해주세요.";
-      passwordInput.classList.add("error");
-      return false;
+      showError(passwordInput, passwordError, "비밀번호를 입력해주세요.");
+      unPasswordValid = false;
     } else if (passwordValue.length < 8) {
-      passwordError.textContent = "비밀번호를 8자 이상 입력해주세요.";
-      passwordInput.classList.add("error");
-      return false;
+      showError(passwordInput, passwordError, "비밀번호를 8자 이상 입력해주세요.");
+      unPasswordValid = false;
+    } else {
+      unPasswordValid = true;
     }
-    return true;
+    toggleLoginButton();
   }
-  // 비밀번호 숨김 상태 확인 함수
-  function togglePasswordVisibility(passwordType, passclosebtn, passopenbtn) {
-    const inPasswordType = passwordType.getAttribute('type') === 'password'; 
 
-    if (inPasswordType) {
-      passwordType.setAttribute('type', 'text'); 
-      passclosebtn.classList.remove('visible');
-      passopenbtn.classList.add('visible');
-    } else {
-      passwordType.setAttribute('type', 'password'); 
-      passclosebtn.classList.add('visible');
-      passopenbtn.classList.remove('visible');
-    }
+  // 비밀번호 숨김 상태 확인 함수
+  function togglePasswordVisibility(passwordType, passwordVisibleButton, passwordInvisibleButton) {
+    const inPasswordType = passwordType.getAttribute('type') === 'password';
+    passwordType.setAttribute('type', inPasswordType ? 'text' : 'password');
```

Comment:
아래처럼 지금보다 더 가독성 있게 변경할 수 있겠어요. 😸

```suggestion
const isPasswordVisible = passwordInput.type === 'text';
passwordInput.type = isPasswordVisible ? 'password' : 'text';
```

Reply:
추가적으로 현재는 예외 케이스가 없겠지만, 늘 예외처리를 생각하고 작성해야해요.
예를들면 아래처럼요.
```
function togglePasswordVisibility(passwordType, passwordVisibleButton, passwordInvisibleButton) {

  // 매개변수 존재 여부 확인
  if (!passwordType || !passwordVisibleButton || !passwordInvisibleButton) {
    console.error('togglePasswordVisibility: 필수 매개변수가 누락되었습니다.');
    return;
  }